### PR TITLE
Frontend/fix: #4482 Enabled feebreakup and fixed booth charges

### PR DESCRIPTION
**Pull request by Rohit4997**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-driver/src/Screens/PaymentHistoryScreen/Transformer.purs [hunk 1]
```diff
@@ -68,12 +68,12 @@ buildTransactionDetails (API.HistoryEntryDetailsEntityV2Resp resp) gradientConfi
                   {
                     key : "FEE_BREAKUP",
                     title : getString FEE_BREAKUP,
-                    val : if (resp.feeType == AUTOPAY_REGISTRATION || not showFeeBreakup) then "" else (getFeeBreakup offerAndPlanDetails driverFee'.totalRides <> " "<> getString GST_INCLUDE)
+                    val : if (resp.feeType == AUTOPAY_REGISTRATION || not showFeeBreakup) then "" else (getFeeBreakup driverFee'.maxRidesEligibleForCharge (driverFee'.planAmount - (fromMaybe 0.0 driverFee'.totalSpecialZoneCharges)) driverFee'.totalRides)
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/PaymentHistoryScreen/Transformer.purs:71]: Here we are subtracting specialZoneCharges from totalPlanAmount. If totalSpecialZoneCharges is nothing then we are assuming that it's zero.

File: Frontend/ui-driver/src/Screens/PaymentHistoryScreen/Transformer.purs [hunk 2]
```diff
@@ -68,12 +68,12 @@ buildTransactionDetails (API.HistoryEntryDetailsEntityV2Resp resp) gradientConfi
...
                   },
                   {
                     key : "BOOTH_CHARGES",
                     title : getString BOOTH_CHARGES,
-                    val : case driverFee'.specialZoneRideCount, driverFee'.specialZoneAmount of
+                    val : case driverFee'.specialZoneRideCount, driverFee'.totalSpecialZoneCharges of
```
Comment (Vignesh-772) [inline Frontend/ui-driver/src/Screens/PaymentHistoryScreen/Transformer.purs:76]: case driverFee'.specialZoneRideCount, driverFee'.totalSpecialZoneCharges of Just 0, Just 0.0 -> "" Just count, Just charges -> show count <> " " <>getString RIDES <> " x ₹" <> getFixedTwoDecimals (charges / INT.toNumber count) <> " " <> getString GST_INCLUDE

if (resp.feeType == AUTOPAY_REGISTRATION || not showFeeBreakup) then "" else (getFeeBreakup driverFee'.maxRidesEligibleForCharge (driverFee'.planAmount - (fromMaybe 0.0 driverFee'.totalSpecialZoneCharges)) driverFee'.totalRides)

move these part to the top let declarations as it is used in two places.

File: Frontend/ui-driver/src/Screens/PaymentHistoryScreen/Transformer.purs [hunk 3]
```diff
@@ -39,6 +39,10 @@ buildTransactionDetails (API.HistoryEntryDetailsEntityV2Resp resp) gradientConfi
                                                   Just (API.DriverFeeInfoEntity driverFee) -> (API.DriverFeeInfoEntity driverFee)
                                                   Nothing -> dummyDriverFee
         statusTime = if resp.feeType == AUTOPAY_PAYMENT then resp.executionAt else resp.createdAt
+        boothCharges specialZoneRideCount totalSpecialZoneCharges = case specialZoneRideCount,totalSpecialZoneCharges of
+                      Just count, Just charges | count /= 0 && charges /= 0.0 -> Just $ show count <> " " <> getString (if count > 1 then RIDES else RIDE) <> " x ₹" <> getFixedTwoDecimals (charges / INT.toNumber count) <> " " <> getString GST_INCLUDE
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/PaymentHistoryScreen/Transformer.purs:43]: Here we are subtracting specialZoneCharges from totalPlanAmount. If totalSpecialZoneCharges is nothing then we are assuming that it's zero.

File: Frontend/ui-driver/src/Screens/PaymentHistoryScreen/Transformer.purs [hunk 4]
```diff
@@ -39,6 +39,10 @@ buildTransactionDetails (API.HistoryEntryDetailsEntityV2Resp resp) gradientConfi
...
+                      _, _ -> Nothing
+        fareBreakup fee = getFeeBreakup fee.maxRidesEligibleForCharge (fee.planAmount - (fromMaybe 0.0 fee.totalSpecialZoneCharges)) fee.totalRides
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/PaymentHistoryScreen/Transformer.purs:45]: Here we are subtracting specialZoneCharges from totalPlanAmount. If totalSpecialZoneCharges is nothing then we are assuming that it's zero.

**Review by Vignesh-772 — APPROVED** [review]

**Review by witcher-shailesh — APPROVED** [review]

**Pull request merged**